### ISSUE TITLE
wxdemo fails to run with wxPython-4.1.2a1.dev5355+6a5e93d0

### ISSUE THREAD

**Issue by BrendanSimon** [wxWidgets/Phoenix#2167]
**Operating system**: OS X 12.4
**wxPython version**:  [4.1.2a1.dev5355+6a5e93d0](https://wxpython.org/Phoenix/snapshot-builds/wxPython-4.1.2a1.dev5355+6a5e93d0-cp39-cp39-macosx_10_10_universal2.whl)
**wxPython-demo version**: [4.1.2a1.dev5355+6a5e93d0](https://wxpython.org/Phoenix/snapshot-builds/wxPython-demo-4.1.2a1.dev5355+6a5e93d0.tar.gz)
**Python version**:  3.9.12

**Description of the problem**:

wxdemo fails to run.  Splash screen appears then an error on the console.

```text
wx._core.wxAssertionError: C++ assertion ""Assert failure"" failed at /Users/robind/projects/bb2/dist-osx-py39/build/ext/wxWidgets/include/wx/withimages.h(188) in GetBitmapBundle(): Image index specified, but there are no images.

Did you forget to call SetImages()?
```

<details>

(venv_py3.9) ➜  demo python demo.py
wx._core.wxAssertionError: C++ assertion ""Assert failure"" failed at /Users/robind/projects/bb2/dist-osx-py39/build/ext/wxWidgets/include/wx/withimages.h(188) in GetBitmapBundle(): Image index specified, but there are no images.

Did you forget to call SetImages()?

The above exception was the direct cause of the following exception:

SystemError: <class 'property'> returned a result with an error set
Traceback (most recent call last):
  File "/Users/brendan/venv_py3.9/lib/python3.9/site-packages/wx/core.py", line 2346, in Notify
    self.notify()
  File "/Users/brendan/venv_py3.9/lib/python3.9/site-packages/wx/core.py", line 3552, in Notify
    self.result = self.callable(*self.args, **self.kwargs)
  File "/Users/brendan/wxPython-demo-4.1.2a1.dev5355+6a5e93d0/demo/Main.py", line 2620, in ShowMain
    frame = wxPythonDemo(None, "wxPython: (A Demonstration)")
  File "/Users/brendan/wxPython-demo-4.1.2a1.dev5355+6a5e93d0/demo/Main.py", line 1560, in __init__
    self.SetOverview(self.overviewText, mainOverview)
  File "/Users/brendan/wxPython-demo-4.1.2a1.dev5355+6a5e93d0/demo/Main.py", line 2153, in SetOverview
    self.nb.SetPageText(0, os.path.split(name)[1])
wx._core.wxAssertionError: C++ assertion ""Assert failure"" failed at /Users/robind/projects/bb2/dist-osx-py39/build/ext/wxWidgets/include/wx/withimages.h(188) in GetBitmapBundle(): Image index specified, but there are no images.

Did you forget to call SetImages()?

</details>

**Comment by brendan-simon-indt** [comment]
It's working now.  I think I may have had the wrong `venv` activated or didn't have the version of wxPython installed that I thought I had.

**Comment by BrendanSimon** [comment]
Oops.  I works on Windows 10, but doesn't work on Mac OS X.
Reopening issue.

**Comment by swt2c** [comment]
I can reproduce the problem.  Strange.  It would be really helpful to have a C++ stacktrace from where that assert is coming from.  For the life of me, I can't get lldb to attach to python.

**Comment by RobinD42** [comment]
You can turn on a mode that will let you see the stacktrace with code like this:

```
self.SetAssertMode(wx.APP_ASSERT_DIALOG|wx.APP_ASSERT_EXCEPTION)
```

Which, in this case gives us this:

![image](https://user-images.githubusercontent.com/808502/172511509-fe8451e4-ff1a-4dcd-a681-96e14798717e.png)

So it looks like things are happening in an order that the mac `wx.Notebook` is not expecting. I'll see if I can untangle things...

**Comment by RobinD42** [comment]
Ah, it's not setting an image list on Macs but is still adding pages with an image id. I guess that was just ignored in the past.

**Comment by RobinD42** [comment]
Fixed in 0295297913c6e7b6624089fc0315e4a3c62629e6

**Comment by BrendanSimon** [comment]
Looks like the build bot failed for this commit, so I couldn't test and verify.
